ci/main.go

```diff
@@ -0,0 +1,211 @@
+	fmt.Println("Building with Dagger")
+
+	// initialize Dagger client
+	// set the current(root) directory on the host as the working directory
+	client, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	// get reference to the local Dockerfile
+	dockerDir := client.Host().Directory("./ci/")
+
+	// initialize image fron dockerfile
+	// mount a host directory in the container at the '/workspaces/parigot' path
+	img := client.Container().Build(dockerDir).WithDirectory("/workspaces/parigot", client.Host().Directory("."),
+		dagger.ContainerWithDirectoryOpts{Exclude: []string{".devcontainer/", "multibuild/", "build/"}}).WithWorkdir("/workspaces/parigot")
+
+	// // get version execute
+	// golang := img.WithExec([]string{"go1.20.4", "version"})
+	// version, err := golang.Stdout(ctx)
+	// if err != nil {
+	// 	return err
+	// }
+	// fmt.Println("Hello from Dagger ", version)
+
+	// // get files in the working directory
+	// contents, err := img.WithExec([]string{"ls", "/src"}).Stdout(ctx)
+	// if err != nil {
+	// 	return (err)
+	// }
+	// fmt.Println(contents)
+
+	img, err = buildProtocGenParigot(ctx, img)
+	if err != nil {
+		return err
+	}
+
+	img, err = generateRep(ctx, img)
+	if err != nil {
+		return err
+	}
+
+	img, err = buildFilePWasm(ctx, img)
+	if err != nil {
+		return err
+	}
+
+	// get reference to build output directory in container
+	output := img.Directory("build")
+
+	// write contents of container build/ directory to the host
+	_, err = output.Export(ctx, "build")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func buildProtocGenParigot(ctx context.Context, img *dagger.Container) (*dagger.Container, error) {
+	/*
+		TEMPLATE=$(shell find command/protoc-gen-parigot -type f -regex ".*\.tmpl")
+		GENERATOR_SRC=$(shell find command/protoc-gen-parigot -type f -regex ".*\.go")
+		build/protoc-gen-parigot: $(TEMPLATE) $(GENERATOR_SRC)
+			@rm -f $@
+			$(GO_TO_HOST) build $(EXTRA_HOST_ARGS) -o $@ github.com/iansmith/parigot/command/protoc-gen-parigot
+	*/
+
+	var err error
+	// template, err = findFilesEndWith("command/protoc-gen-parigot", ".tmpl")
+	// if err != nil {
+	// 	return img, err
+	// }
+	// eneratorSrc, err = findFilesEndWith("test", ".go")
+	// if err != nil {
+	// 	return img, err
+	// }
+	template, err = img.WithExec([]string{"bash", "-c", `find command/protoc-gen-parigot -type f -regex ".*\.tmpl"`}).Stdout(ctx)
+	if err != nil {
+		return img, err
+	}
+	eneratorSrc, err = img.WithExec([]string{"bash", "-c", `find command/protoc-gen-parigot -type f -regex ".*\.go"`}).Stdout(ctx)
+	if err != nil {
+		return img, err
+	}
+
+	// contents, err := img.WithExec([]string{"ls", "-l", "."}).Stdout(ctx)
+	// if err != nil {
+	// 	return img, err
+	// }
+	// fmt.Println(contents)
+
+	target := "build/protoc-gen-parigot"
+	packagePath := "github.com/iansmith/parigot/command/protoc-gen-parigot"
+	img = img.WithExec([]string{"rm", "-f", target})
```

Comment:
If you like doing the  `rm` first, I'm cool with it....  But my guess is that the `rm` is not needed because  dagger's `WithExec`  will  probably overwrite the previous version... but that's a guess.

Reply:
I think the overwrite depends on the cmd, because WithExec just executes the cmd inside it.

When I run "go build", it overwrites the previously generated executable with the same name, right?